Sources/TuistKit/Embed/FrameworkEmbedder.swift

```diff
@@ -63,12 +61,25 @@ final class FrameworkEmbedder: FrameworkEmbedding {
             try fileHandler.createFolder(productFrameworksPath)
         }
 
+        if let codeSigningIdentity = environment.codeSigningIdentity {
+            try codesignFramework(frameworkAbsolutePath: frameworkAbsolutePath, codeSigningIdentify: codeSigningIdentity)
+        }
+        
         try copyFramework(productFrameworksPath: productFrameworksPath, frameworkAbsolutePath: frameworkAbsolutePath, validArchs: validArchs)
         try copySymbols(frameworkDsymPath: frameworkDsymPath, destinationPath: destinationPath, validArchs: validArchs)
         try copyBCSymbolMaps(action: action, frameworkAbsolutePath: frameworkAbsolutePath, builtProductsDir: builtProductsDir)
     }
 
     // MARK: - Fileprivate
+    
+    private func codesignFramework(frameworkAbsolutePath: AbsolutePath, codeSigningIdentify: String) throws {
+        
+        try system.run([
+            "/usr/bin/xcrun",
+            "codesign", "--force", "--sign", codeSigningIdentify, "--preserve-metadata=identifier,entitlements", frameworkAbsolutePath.pathString
```

Comment:
Would you mind adding a comment that explains why the argument `--preserve-metadata=identifier,entitlements` is necessary?

Reply:
It's required because if the framework is already code signed we don't want to override the original identifier or entitlements (as this may potentially break your setup) - I will add this as a comment.

**man codesign**
``` 
    --preserve-metadata=list
             When re-signing code that is already signed, reuse some information
             from the old signature.  If new data is specified explicitly, it is
             preferred.  You still need to specify the -f (--force) option to
             enable overwriting signatures at all.  If this option is absent,
             any old signature has no effect on the signing process.
             This option takes a comma-separated list of names, which you may
             reasonably abbreviate:

             identifier    Preserve the signing identifier (--identifier)
                           instead of generating a default identifier.

             entitlements  Preserve the entitlement data (--entitlements).
```